File: 242-ValidAnagram/242-ValidAnagram.py
# Last updated: 9/27/2025, 11:23:52 AM
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        dct1={}
        for i in s:
            if i not in dct1.keys():
                dct1[i]=1
            else:
                dct1[i]+=1
        dct2={}
        for i in t:
            if i not in dct2.keys():
                dct2[i]=1
            else:
                dct2[i]+=1
        for i in dct2.keys():
            if i not in dct1.keys():
                return False
            else:
                if dct1[i]!=dct2[i]:
                    return False
        if len(s)!=len(t):
            return False
        return True

File: 242-ValidAnagram/242-ValidAnagram.cpp
// Last updated: 4/4/2026, 12:42:02 AM
class Solution {
public:
    bool isAnagram(string s, string t) {
        int arrs[26];
        int arr[26];
        memset(arr,0,26);
        memset(arrs,0,26);
        for (int i=0;i<s.length();i++){
            arrs[(s[i]) -'a']+=1;
        }
        for (int j=0;j<t.length();j++){
            arr[(t[j])-'a']+=1;
        }
        bool fnd=true;
        if (s.length()==t.length()){
            for (int i=0;i<26;i++){
                if (arr[i]!=arrs[i]){
                    fnd=false;
                    break;
                }
            }
            
        }
        else{
            fnd=false;
        }
        return fnd;
    }
};